#hw02_fix_app

DMA30 opened the pull request.

Commit:
#hw02_fix_app

vladislavmass commented on hw02_fix_app/go.mod:
```diff
@@ -1,5 +1,3 @@
 module github.com/fixme_my_friend/hw02_fix_app
```
Неверное имя модуля. Верное имя модуля в данном случае `github.com/DMA30/DMA/hw02_fix_app`

vladislavmass commented on hw02_fix_app/reader/pkg.go:
```diff
@@ -24,6 +25,11 @@ func ReadJSON(filePath string, limit int) ([]types.Employee, error) {
 
+	if err != nil {
+		fmt.Printf("Error: %v", err)
+		return nil, nil
```
Тут возврат должен иметь вид `return nil,err`

vladislavmass commented on hw02_fix_app/reader/pkg.go:
```diff
@@ -24,6 +25,11 @@ func ReadJSON(filePath string, limit int) ([]types.Employee, error) {
+		return nil, nil
+	}
+
 	res := data
```
Эта строчка ненужна. Можно сразу возвращать `data`

hungry-honeybadger reviewed (changes requested):
Хорошее начало. Работу принять не могу так как неправильное имя модуля в go.mod

Commit:
(refs hw02_fix_app) Чиним, собираем и запускаем приложение

*go.mod - исправлены ошибки
*reader/pkg.go - исправлены ошибки

Commit:
(refs hw02_fix_app) Чиним, собираем и запускаем приложение

*main.go - исправлены ошибки
*printer/pkg.go - исправлены ошибки
*reader/pkg.go - исправлены ошибки

hungry-honeybadger commented on hw02_fix_app/main.go:
```diff
@@ -18,10 +19,8 @@ func init() {
-	staff, err = reader.ReadJSON(path, -1)
+	staff, err = reader.ReadJSON(path)
 
 	fmt.Print(err)
```
Тут нужен блок if. Если err не равен nil, печатаем ошибку иначе печатаем результат

hungry-honeybadger commented on hw02_fix_app/printer/pkg.go:
```diff
@@ -1,13 +1,17 @@
 	var str string
 	for i := 0; i < len(staff); i++ {
-		str := fmt.Sprintf("User ID: %d; Age: %d; Name: %s; Department ID: %d; ", staff[i].UserID, staff[i].Age, staff[i].Name, staff[i].DepartmentID)
+		str = fmt.Sprintf("User ID: %d; Age: %d; Name: %s; Department ID: %d; ",
```
тут хватило бы `fmt.Println(staff[i])`

hungry-honeybadger commented on hw02_fix_app/reader/pkg.go:
```diff
@@ -1,20 +1,21 @@
+func ReadJSON(filePath string) ([]types.Employee, error) {
 	f, err := os.Open(filePath)
 	if err != nil {
 		fmt.Printf("Error: %v", err)
```
Тут должен быть возврат ошибки

hungry-honeybadger commented on hw02_fix_app/reader/pkg.go:
```diff
@@ -1,20 +1,21 @@
+	bytes, err := io.ReadAll(f)
 	if err != nil {
 		fmt.Printf("Error: %v", err)
 		return nil, nil
```
Тут должен быть возврат ошибки
